build.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+clear
+dirpath=$(dirname "$(readlink -f "$0")")
+
+# Check if the virtual environment exists
+if [ ! -d "$dirpath/env" ]; then
+    echo
+    echo "No virtual environment found! Run setup_env.sh to set it up first."
+    echo
+    read -p "Press any key to continue..."
```

Comment:
Is there a distro or file manager out there that lets us double-click `.sh` scripts to run them? I don't remember any that allows that by default...

If there's not, I think we could reduce friction and remove this and other calls to `read -p`.

I use KDE Plasma, and in Dolphin (their default file manager), they have this feature through the right-click menu (*Run In Konsole*), but even then, once the script finishes, they keep the terminal window open.

Reply:
Nope, tho we might be able to make a workaround and just run the script directly? it would definitely make it much easier

```sh
# Check if the virtual environment exists
if [ ! -d "$dirpath/env" ]; then
    echo
    echo "No virtual environment found! Running setup_env.sh to set it up first."
    echo
    "$dirpath/setup_env.sh"
    if [ $? -ne 0 ]; then
        echo "Failed to set up the virtual environment."
        exit 1
    fi
fi
```

Reply:
Sorry, my comment above this one was not clear.

I asked that because I assumed that the only reason you're using `read -p` to wait for user input, is in a scenario where users would run the script by double-clicking on them, and therefore those `read -p` calls would prevent the spawned terminal window from being closed upon script completion.

The Windows `.bat` scripts have this as well (through `pause` calls), to avoid command prompt window from being closed automatically too.

However, this is usually not done with Linux scripts, because most people will open an empty terminal window first, and then run the script later...

Reply:
ohh yeah sorry for misunderstanding, i do sometimes run the scripts by clicking on it and selecting "Execute in terminal", tho as you said most people open a clean terminal and then run it
![image](https://github.com/user-attachments/assets/900b271f-9b76-4160-baeb-ae331db09daf)

Personally i think it's best to keep it, sure it's an extra step for those opening a clear terminal but i can definitely see it creating issues for new people opening them the other way

Reply:
Now the question is just, would it be better do keep it that way or add that it opens the script itself, personally do i think it would be nice, but i can also see why you would keep them separate

Reply:
Yeah, it's fine to keep them, I just didn't know of a file manager that could do that. :p

Out of curiosity, if you choose "Execute in Terminal", will that file manager automatically close the spawned terminal window when the script finishes running?

Reply:
Yes! function wise does it work exactly like when you open a batch script by double clicking on it

Reply:
>Now the question is just, would it be better do keep it that way or add that it opens the script itself, personally do i think it would be nice, but i can also see why you would keep them separate

I think it's better to keep these two scripts just for documentation purposes.

Since Windows also uses two scripts, we avoid having to rewrite parts of the Wiki.